mkosi/backend.py

```diff
@@ -420,7 +418,9 @@ def __post_init__(self) -> None:
 
     @property
     def root(self) -> Path:
-        return self.workspace / "root"
+        p = self.workspace / "root"
+        p.mkdir(exist_ok=True, mode=0o755)
```

Comment:
I'm not entirely happy with a property having a side effect, just because that's not what one usually expects from attribute lookup. Maybe this could become a method or grow a new method `ensure_root_exists` or similar, that is called once somewhere at the beginning.